packages/button/src/styled/Button.tsx

```diff
@@ -25,7 +25,9 @@ ${({ css: cssInner }) => (cssInner || null)}
 direction: ${({ isRTL }) => (isRTL ? 'rtl' : 'ltr')};
 `;
 
-export const Root = (tag: string) => styled[tag]`
+export const Root = (tag: string | React.ElementType) => (typeof tag === 'string'
+  ? styled[tag]
+  : styled(tag))`
   align-items: center;
   border-width: 0;
```

Comment:
border-width property unnecessary because of border property below.
![image](https://user-images.githubusercontent.com/17886232/69966300-c403b180-151e-11ea-904c-648f99e1d338.png)